src/main/scala/dpla/ingestion3/reports/PropertyValueReport.scala

```diff
@@ -0,0 +1,108 @@
+package dpla.ingestion3.reports
+
+import org.apache.spark.sql.{DataFrame, Row, SparkSession, Dataset}
+import dpla.ingestion3.model._
+
+/**
+  * Property Value QA report.  Takes one field and gives a count of unique
+  * values.
+  *
+  * Fields currently supported:
+  *   - sourceResource.format
+  *   - sourceResource.rights
+  *   - sourceResource.type
+  *
+  * @example Example invocation:
+  *
+  *   $ sbt "run-main dpla.ingestion3.ReporterMain \
+  *     /path/to/enriched-data.avro /path/to/propvalreport local[2] \
+  *     propertyValue sourceResource.type"
+  *
+  *     ... This produces a directory named 'propvalreport' with a file in it
+  *     named 'part-< ... >.csv', with contents like:
+  *
+  *     value,count
+  *     physicalobject,1
+  *     dataset,13
+  *     [...]
+  *
+  * @param inputURI         Input URI or file path (Avro file / directory)
+  * @param outputURI        Output URI or file path (directory containing CSV
+  *                         file)
+  * @param sparkMasterName  Spark master name, e.g. "local[1]"
+  * @param params           Additional parameters, currently:
+  *                         params(0): The DPLA MAP field to analyze
+  */
+class PropertyValueReport (
+                            val inputURI: String,
+                            val outputURI: String,
+                            val sparkMasterName: String,
+                            val params: Array[String]
+                          ) extends Report {
+
+  /*
+   * We set instance fields from constructor arguments, and override accessor
+   * methods in order to make the class more amenable to unit testing.
+   * However ...
+   * FIXME: It's not clear how to write unit tests that involve Dataframes
+   * and Datasets, so there are no such tests for this class.
+   *
+   */
+  override val sparkAppName: String = "PropertyValueReport"
+  override def getInputURI: String = inputURI
```

Comment:
I think I considered that, but wanted them to fail hard in the case of no input file, etc.